internal/controllers/topology/cluster/conditions.go

```diff
@@ -139,6 +140,32 @@ func (r *Reconciler) reconcileTopologyReconciledCondition(s *scope.Scope, cluste
 
 	// If any of the lifecycle hooks are blocking any part of the reconciliation then topology
 	// is not considered as fully reconciled.
+	if feature.Gates.Enabled(feature.RuntimeSDK) {
+		if annotations.HasWithPrefix(clusterv1.BeforeClusterUpgradeHookAnnotationPrefix, cluster.Annotations) {
+			var annotation string
+			for k, v := range cluster.Annotations {
+				if strings.HasPrefix(k, clusterv1.BeforeClusterUpgradeHookAnnotationPrefix) {
+					annotation = fmt.Sprintf("%s=%s", k, v)
```

Comment:
I would only include the key (see my suggestion for the log).

I think we mostly assume the value should be empty and not contain additional info